### PR TITLE
Modularization fixes for `fast_callable` interpreters

### PR DESCRIPTION
**Pull request by mkoeppe**
### :books: Description

The code generator in `sage_setup.autogen.interpreters` can now build a subset of interpreters. This is used in #35095 to only build the interpreters for `Element` and `object` in the sagemath-categories distribution.

At runtime, `sage.ext.fast_callable` no longer fails when an interpreter cannot be imported but falls back to the `Element` interpreter if necessary.

- Part of: #29705 
- Cherry-picked from: #35095 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage_setup/autogen/interpreters/__init__.py [hunk 1]
```diff
@@ -174,7 +159,7 @@ def build_interp(interp_spec, dir):
         write_if_changed(path, method())
 
 
-def rebuild(dirname, force=False):
+def rebuild(dirname, force=False, interpreters=None, distribution=None):
```
Comment (orlitzky) [inline src/sage_setup/autogen/interpreters/__init__.py:163]: If you're adding arguments, can you add the missing `INPUT` block that explains what they do?

File: src/sage/ext/fast_callable.pyx [hunk 2]
```diff
@@ -465,41 +465,73 @@ def fast_callable(x, domain=None, vars=None,
...
-                                len(vars),
-                                domain)
+    builder, str = _builder_and_stream(vars=vars, domain=domain)
+
     generate_code(et, str)
     str.instr('return')
     return builder(str.get_current())
 
+
+def _builder_and_stream(vars, domain):
```
Comment (orlitzky) [inline src/sage/ext/fast_callable.pyx:475]: I can see what's going on from the diff, but if someone comes along a year from now, this docstring isn't really going to explain what the function is doing and why. Can you  expand it a little?

Also, if possible, can you test the import-failure code path?

Comment (mkoeppe) [reply src/sage/ext/fast_callable.pyx:475]: This was just mechanical refactoring of code that was nested a bit too deep. Let me see if I remember what it is doing, after a mere 4 days, not a year...

File: src/sage_setup/autogen/interpreters/__init__.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage_setup/autogen/interpreters/__init__.py:163]: done in 12c04b8

File: src/sage/ext/fast_callable.pyx (hunk 2 above)
Comment (mkoeppe) [reply src/sage/ext/fast_callable.pyx:475]: > can you test the import-failure code path?

done in afd3be6

File: src/sage_setup/autogen/interpreters/specs/rdf.py [hunk 3]
```diff
@@ -60,14 +61,14 @@ def __init__(self):
 
         Make sure that pow behaves reasonably::
 
-            sage: var('x,y')
+            sage: var('x,y')                                                            # optional - sage.symbolic
```
Comment (orlitzky) [inline src/sage_setup/autogen/interpreters/specs/rdf.py]: This should avoid the optional tag:

```
from sage.ext.fast_callable import ExpressionTreeBuilder
etb = ExpressionTreeBuilder(vars=('x','y'))
x = etb.var('x')
y = etb.var('y')
ff = fast_callable(x^y, domain=RDF)
```

Comment (mkoeppe) [reply src/sage_setup/autogen/interpreters/specs/rdf.py]: thanks, done in 97135ad

**Review by orlitzky — APPROVED** [review]
Ok, presumably you know what's going on with `all_py`, and the rest LGTM.

**Comment by mkoeppe** [comment]
Thank you! May I set to "positive review" then?

**Comment by orlitzky** [comment]
Yes, it'll take me a few years to remember to do it.

**Pull request merged**